### ISSUE TITLE
onSound triggers constantly if playing melody on the spseaker

### ISSUE THREAD

**Issue by microbit-mark** [microsoft/pxt-microbit#3777]
**Describe the bug**

This program triggers the sound playback constantly

<img width="479" alt="image" src="https://user-images.githubusercontent.com/31242877/100466052-34102d80-30c8-11eb-85b9-f0bba1c2c64e.png">

Setting a higher threshold resolves this
<img width="371" alt="image" src="https://user-images.githubusercontent.com/31242877/100466130-543fec80-30c8-11eb-8849-73fa02864563.png">

But `set threshold` is not a kid-friendly block for first using the microphone and it's hidden away in Input > more

Could we change the default threshold so it does not trigger so easily.

...or  we could have a mic on/off block
... or lower the volume of the melody

**Comment by pelikhan** [comment]
We should auto turn off the microphone when I paging a sound

________________________________

**Comment by jaustin** [comment]
@finneyj is there an easy way just to ignore mic events during sound playing? I would rather not actually turn off the microphone

**Comment by finneyj** [comment]
Sounds like a "learning opportunity" about feedback loops.. :)

We could suppress the generation of events, for sure... but I think there's a problem here around making too many assumptions... Kids won't expect the microphone to be enabled/disabled without them knowing.

There could be use cases where disabling the mic events could break kids code. e.g. clap to stop sound? Music playing in a game, and a clap to activate a game activity? etc...  

Alternative options:

- Preserve the current behaviour and encourage kids to manually set threshold block anyway. Can we do tooltips if we see a block that makes a sound inside an "onLoud" handler? 

- Try to dynamically adjust the onLoud threshold based on sound output level. We calibrate for this (as all micro:bit are more or less created equal), but only for sound that go through the mixer (anyone sneaking around that to do old school v1 PWM sounds wouldn't work). Also if kids/teachers were using external speakers, that calibration would be no good, and you'll likely still get the same feedback loop. This would be less of a common case though I guess.

- Add some sort of feedback suppression filter in software, like you do on your phone/laptop. I don't think we have the developer resource or CPU cycles to do that though.

**Comment by pelikhan** [comment]
A microphone mute block would match perfectly what they already use in Zoom/Teams/… every day.

**Comment by finneyj** [comment]
yep - mute would work.

**Comment by jaustin** [comment]
Right, Joe and John implemented a function in CODAL to enable this. CODAL now has an 'is sound playing' 

https://github.com/lancaster-university/codal-microbit-v2/commit/cc02e7a54b9332cccca2a2af98eb10621c52e693

We should expose this as a block to users!

So then the question is should we still have a 'microphone mute' block as well or should we add a new sensitivity level to the microphone (set microphone sensitivity should be coming to '`input` as part of #4493

**Comment by jaustin** [comment]
@abchatra this one is still a good issue and we now have the stuff we need in CODAL to address it. It becomes more relevant/prevalent with the audio record/playback stuff

**Comment by jaustin** [comment]
@abchatra any more thoughts on adding this?

**Comment by abchatra** [comment]
I had already marked it for release. Someone will pick it up during bug fixing phase.

**Comment by jaustin** [comment]
Great, and sorry, was looking for a 'next release' label, not the milestones. Oops. (looks like that was what we used last year!)

**Comment by abchatra** [comment]
@srietkerk can you take this up? Basically we need to expose  a new block **"sound is playing"** in Music category under Volume subcategory. Whenever a sound is playing this should return 1 other wise 0.

**Comment by srietkerk** [comment]
Yep, sounds good!
